### PR TITLE
Explicitly catch and propagate signals to subcommands #minor

### PR DESCRIPTION
**Pull request by sbogacz**

### CODE REVIEW

File: cmd/aws-env/main.go [hunk 1]
```diff
@@ -214,7 +216,43 @@ func invoke(r *awsenv.Replacer, prog string, args []string) error {
...
+	// wait for the command to finish
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- cmd.Wait()
+		close(errCh)
+	}()
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh)
+
+	for {
```
Comment (mrobinsn) [inline cmd/aws-env/main.go:238]: This loop isn't necessary, every case ends with a `return`.

Comment (mrobinsn) [reply cmd/aws-env/main.go:238]: nevermind, this so that we pass along all signals even non terminating ones?

**Comment by mrobinsn** [comment]
I changed this to a minor release since we are changing the way signals are handled, seems like more than a patch change.

File: cmd/aws-env/main.go (hunk 1 above)
Comment (sbogacz) [reply cmd/aws-env/main.go:238]: Correct

**Comment by sbogacz** [comment]
> I changed this to a minor release since we are changing the way signals are handled, seems like more than a patch change.

From semver.org:

```
Given a version number MAJOR.MINOR.PATCH, increment the:

MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards compatible manner, and
PATCH version when you make backwards compatible bug fixes.
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
```

I'm inclined to say that this is more bug fix than "adding functionality"

**Comment by mrobinsn** [comment]
I would call this new functionality, and I could almost be convinced that its not backwards compatible if you were relying on the signal handle to work how it used to (even without knowing it).

**Review by mrobinsn — APPROVED** [review]

**Pull request merged**